osu.Game/Rulesets/Mods/ModMetronome.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics.Containers;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.UI;
+using osu.Game.Skinning;
+using osu.Game.Configuration;
+using osu.Framework.Bindables;
+using osu.Game.Audio;
+using osu.Game.Beatmaps.Timing;
+using osu.Framework.Graphics;
+using System.Linq;
+using System;
+
+namespace osu.Game.Rulesets.Mods
+{
+    public abstract class ModMetronome : Mod
+    {
+        public override string Name => "Metronome";
+        public override string Acronym => "MT";
+        public override IconUsage? Icon => FontAwesome.Solid.TachometerAlt;
+        public override ModType Type => ModType.DifficultyReduction;
+        public override string Description => "Add a metronome";
+        public override bool Ranked => false;
+        public override double ScoreMultiplier => 0.0;
+
+        public override Type[] IncompatibleMods => base.IncompatibleMods.Append(typeof(ModNightcore)).ToArray();
+
+        public enum TickFrequency
+        {
+            One = 1,
+            Two = 2,
+            Four = 4
+        }
+
+        [SettingSource("Tick frequency", "Number of metronome ticks per beat")]
```

Comment:
i don't understand this setting. it's not actually doing what it says (and only delays the "accent" beat more?)